controller/src/test/java/io/pravega/controller/store/stream/ZKStreamMetadataStoreTest.java

```diff
@@ -130,34 +133,95 @@ public void testScaleMetadata() throws Exception {
         assertTrue(scaleIncidents.size() == 3);
         assertTrue(scaleIncidents.get(0).getSegments().size() == 2);
         assertTrue(scaleIncidents.get(1).getSegments().size() == 2);
 
         // scale again
-        scale(scope, stream, scaleIncidents.get(0).getSegments());
+        scale(scope, stream, scaleIncidents.get(0).getSegments(), newRanges);
         scaleIncidents = store.getScaleMetadata(scope, stream, null, executor).get();
         assertTrue(scaleIncidents.size() == 4);
         assertTrue(scaleIncidents.get(0).getSegments().size() == 2);
         assertTrue(scaleIncidents.get(1).getSegments().size() == 2);
     }
 
-    private void scale(String scope, String stream, List<Segment> segments) {
 
-        AbstractMap.SimpleEntry<Double, Double> segment1 = new AbstractMap.SimpleEntry<>(0.0, 0.5);
-        AbstractMap.SimpleEntry<Double, Double> segment2 = new AbstractMap.SimpleEntry<>(0.5, 1.0);
+    @Test
+    public void testSplitsMerges() throws Exception {
+        String scope = "testScopeScale";
+        String stream = "testStreamScale";
+        ScalingPolicy policy = ScalingPolicy.fixed(2);
+        StreamConfiguration configuration = StreamConfiguration.builder().scope(scope).streamName(stream).scalingPolicy(policy).build();
+
+        store.createScope(scope).get();
+        store.createStream(scope, stream, configuration, System.currentTimeMillis(), null, executor).get();
+        store.setState(scope, stream, State.ACTIVE, null, executor).get();
+
+        // Case: Initial state, splits = 0, merges = 0
+        // time t0, total segments 2, S0 {0.0 - 0.5} S1 {0.5 - 1.0}
+        List<ScaleMetadata> scaleIncidents = store.getScaleMetadata(scope, stream, null, executor).get();
+        assertTrue(scaleIncidents.size() == 1);
+        org.apache.commons.lang3.tuple.Pair<Long, Long> pairSplitsMerges = store.findNumSplitsMerges(scope, stream, executor).get();
+        assertEquals("Number of splits ", new Long(0), pairSplitsMerges.getLeft());
+        assertEquals("Number of merges", new Long(0), pairSplitsMerges.getLeft());
+
+        // Case: Only splits, S0 split into S2, S3, S4 and S1 split into S5, S6,
+        //  total splits = 2, total merges = 3
+        // time t1, total segments 5, S2 {0.0, 0.2}, S3 {0.2, 0.4}, S4 {0.4, 0.5}, S5 {0.5, 0.7}, S6 {0.7, 1.0}
+        AbstractMap.SimpleEntry<Double, Double> segment2 = new AbstractMap.SimpleEntry<>(0.0, 0.2);
+        AbstractMap.SimpleEntry<Double, Double> segment3 = new AbstractMap.SimpleEntry<>(0.2, 0.4);
+        AbstractMap.SimpleEntry<Double, Double> segment4 = new AbstractMap.SimpleEntry<>(0.4, 0.5);
+        AbstractMap.SimpleEntry<Double, Double> segment5 = new AbstractMap.SimpleEntry<>(0.5, 0.7);
+        AbstractMap.SimpleEntry<Double, Double> segment6 = new AbstractMap.SimpleEntry<>(0.7, 1.0);
+        List<java.util.AbstractMap.SimpleEntry<Double, Double>> newRanges1 = Arrays.asList(segment2, segment3, segment4, segment5, segment6);
```

Comment:
I'm not sure why you have the full qualified class name here. Please use imports.